### PR TITLE
feat: adding support for checkbox filter

### PR DESCRIPTION
**Pull request by anandtiwary**
## Description
Please include a summary of the change, motivation and context.
Adding support for adding checkbox filter to table widgets

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

**Comment by anandtiwary** [comment]
Looking for early feedback. I think this needs more work. I need to provide graphql filter for both checked and unchecked state. 
```
{
  type: 'checkbox-table-widget-filter',
  checked: boolean,
  checkedFilter: TableWidgetFilterMode;
  uncheckedFilter: TableWidgetFilterMode
}
```

File: projects/components/src/table/controls/table-controls.component.ts [hunk 1]
```diff
@@ -62,6 +71,19 @@ export class TableControlsComponent implements OnChanges {
   @Input()
   public activeModeItem?: ToggleItem;
 
+  // Checkbox filter
+  @Input()
+  public showCheckbox?: boolean;
```
Comment (aaron-steinfeld) [inline projects/components/src/table/controls/table-controls.component.ts:76]: Is this implicit via the other checkbox inputs being provided?

Comment (anandtiwary) [reply projects/components/src/table/controls/table-controls.component.ts:76]: Yes. Currently, if checkbox filter model is present and is valid, then the renderer will set this input

Comment (aaron-steinfeld) [reply projects/components/src/table/controls/table-controls.component.ts:76]: I was thinking we could move that implicitness down a level - if a label or boolean is provided, then we should show it. Else hide it. No need for the extra input

Comment (anandtiwary) [reply projects/components/src/table/controls/table-controls.component.ts:76]: I guess I can make checkboxLabel a required property and hide the filter if it is missing

Comment (aaron-steinfeld) [reply projects/components/src/table/controls/table-controls.component.ts:76]: I think that makes sense - we'd never have a checkbox floating without a label (I sure hope...)

**Review by aaron-steinfeld — DISMISSED** [review]

**Review by aaron-steinfeld — COMMENTED** [review]
This combined in a different branch that's unrelated, please revert

**Comment by anandtiwary** [comment]
sorry. reverted the unrelated commit

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**